src/app/app.js

```diff
@@ -1,18 +1,41 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
 import { BrowserRouter as Router } from 'react-router-dom';
+import PropTypes from 'prop-types';
 import Header from './header';
 import RoutesResolver from './routes/index';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import 'antd/dist/antd.css';
 import '../index.css';
+import controller from './controller/TelegramController';
+import { getAuthorizedState } from './redux/selector';
+import { initAuthorizeState } from './redux/authorizationAction';
+
+const App = ({ authorized, dispatchAuthorizeState }) => {
+  useEffect(() => {
+    controller.init();
+    dispatchAuthorizeState();
+  }, [dispatchAuthorizeState]);
 
-const App = () => {
   return (
     <Router>
-      <Header />
-      <RoutesResolver />
+      {authorized && <Header />}
```

Comment:
I know that your variant works, but it is better to do conditional rendering in the following way: 
`{authorized ? <Header /> : null}`
Explicitly returning null will guarantee that you don't see unexpected values like **undefined** on screen. In case if **authorized** will get undefined occasionally.